src/python/pose_format/utils/generic.py

```diff
@@ -62,31 +61,50 @@ def normalize_pose_size(pose: Pose, target_width: int = 512):
     pose.header.dimensions.height = pose.header.dimensions.width = target_width
 
 
-def pose_hide_legs(pose: Pose):
+def pose_hide_legs(pose: Pose, remove: bool = False) -> Pose:
+    """
+    Hide or remove leg components from a pose.
+    
+    If `remove` is True, the leg components are removed; otherwise, they are hidden (zeroed out).
+    """
     known_pose_format = detect_known_pose_format(pose)
+    
     if known_pose_format == "holistic":
-        point_names = ["KNEE", "ANKLE", "HEEL", "FOOT_INDEX"]
-        # pylint: disable=protected-access
-        points = [
-            pose.header._get_point_index("POSE_LANDMARKS", side + "_" + n)
-            for n in point_names
-            for side in ["LEFT", "RIGHT"]
-        ]
-        pose.body.data[:, :, points, :] = 0
-        pose.body.confidence[:, :, points] = 0
+        point_names = ["KNEE", "ANKLE", "HEEL", "FOOT_INDEX", "HIP"]
+        sides = ["LEFT", "RIGHT"]
+        point_names_to_remove = [f"{side}_{name}" for side in sides for name in point_names]
+        points_to_remove_dict = {
+            "POSE_LANDMARKS": point_names_to_remove,
+            "POSE_WORLD_LANDMARKS": point_names_to_remove,
+        }
+
     elif known_pose_format == "openpose":
         point_names = ["Hip", "Knee", "Ankle", "BigToe", "SmallToe", "Heel"]
-        # pylint: disable=protected-access
-        points = [
-            pose.header._get_point_index("pose_keypoints_2d", side + n) for n in point_names for side in ["L", "R"]
-        ]
-        pose.body.data[:, :, points, :] = 0
-        pose.body.confidence[:, :, points] = 0
+        sides = ["L", "R", "Mid"]
+        point_names_to_remove = [f"{side}{name}" for side in sides for name in point_names]
+        points_to_remove_dict = {"pose_keypoints_2d": point_names_to_remove}
+
     else:
         raise NotImplementedError(
             f"Unsupported pose header schema {known_pose_format} for {pose_hide_legs.__name__}: {pose.header}"
         )
 
+    if remove:
+        return pose.remove_components([], points_to_remove_dict)
+
+    # Hide the points instead of removing them
+    # Some of the generated point_names_to_remove don't exist, e.g. MidHip, so get_point_index gives None
+    points = []
+    for name in point_names_to_remove:
```

Comment:
it would make more sense to me to iterate more on `points_to_remove_dict`
```py
for component, points in points_to_remove_dict.items():
    for point_name in points:
       point_index = pose.header.get_point_index(component, point_name)
```